linera-faucet/server/src/lib.rs

```diff
@@ -322,32 +329,47 @@ where
         }
     }
 
-    /// Executes a batch of chain creation requests
+    /// Executes a batch of chain creation requests.
     async fn execute_batch(
         &mut self,
         batch_requests: Vec<PendingRequest>,
     ) -> Result<(), anyhow::Error> {
-        // Pre-validate: check rate limiting and existing chains
+        // Pre-validate: check existing chains.
         let mut valid_requests = Vec::new();
 
         for request in batch_requests {
-            // Check if this owner already has a chain
-            {
-                let storage = self.faucet_storage.lock().await;
-                if let Some(existing_description) = storage.get_chain(&request.owner) {
-                    let _ = request.responder.send(Ok(existing_description.clone()));
+            // Check if this owner already has a chain. Otherwise send response immediately.
+            let response = match self.faucet_storage.get_chain_id(&request.owner).await {
+                Ok(None) => {
+                    valid_requests.push(request);
                     continue;
                 }
+                Ok(Some(existing_chain_id)) => {
+                    // Retrieve the chain description from local storage.
+                    get_chain_description_from_storage(
+                        self.client.storage_client(),
+                        existing_chain_id,
+                    )
+                    .await
+                }
+                Err(err) => {
+                    tracing::error!("Database error: {err}");
+                    Err(Error::new(err.to_string()))
+                }
+            };
+            if let Err(response) = request.responder.send(response) {
+                tracing::error!(
+                    "Receiver dropped while sending response {response:?} for request from {}",
+                    request.owner
+                );
```

Comment:
Why not put this inside the `Ok(Some(existing_chain_id))` ? It'd be clearer that this snippet is executed only for already-existing chains.

Reply:
I wanted to deduplicate the `send` call and `error!` log: We send a response in both the `Ok(Some(_))` and `Err(_)` cases.